taskcluster/fenix_taskgraph/transforms/visual_metrics.py

```diff
@@ -0,0 +1,84 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+Generate labels for tasks without names, consistently.
+Uses attributes from `primary-dependency`.
+"""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os
+
+from taskgraph.transforms.base import TransformSequence
+
+transforms = TransformSequence()
+
+SYMBOL = "%(groupSymbol)s(%(symbol)s-vismet)"
+# the test- prefix makes the task SETA-optimized.
+LABEL = "test-vismet-%(platform)s-%(label)s"
+
+
+@transforms.add
+def make_label(config, jobs):
+	""" Generate a sane label for a new task constructed from a dependency
+	Using attributes from the dependent job and the current task kind"""
+	for job in jobs:
+		dep_job = job['primary-dependency']
+		attr = dep_job.attributes.get
+
+		if attr('locale', job.get('locale')):
+			template = "{kind}-{locale}-{build_platform}/{build_type}"
+		elif attr('l10n_chunk'):
+			template = "{kind}-{build_platform}-{l10n_chunk}/{build_type}"
+		elif config.kind.startswith("release-eme-free") or \
+				config.kind.startswith("release-partner-repack"):
+			suffix = job.get("extra", {}).get("repack_suffix", None) or \
+					 job.get("extra", {}).get("repack_id", None)
+			template = "{kind}-{build_platform}"
+			if suffix:
+				template += "-{}".format(suffix.replace('/', '-'))
+		else:
+			template = "{kind}-{build_platform}/{build_type}"
+		job['label'] = template.format(
+			kind=config.kind,
+			build_platform=attr('build_platform'),
+			build_type=attr('build_type'),
+			locale=attr('locale', job.get('locale', '')),  # Locale can be absent
+			l10n_chunk=attr('l10n_chunk', '')  # Can be empty
+		)
+
+		yield job
+
+
+@transforms.add
+def run_visual_metrics(config, jobs):
+	for job in jobs:
+		dep_job = job.pop('primary-dependency', None)
+		if dep_job is not None:
+			platform = dep_job.task['extra']['treeherder-platform']
+			job['dependencies'] = {dep_job.label: dep_job.label}
+
+			# Add the artifact to be processed as a fetches artifact
+			job['fetches'][dep_job.label] = []
+			job['fetches'][dep_job.label].append({
+				'artifact': 'browsertime-results.tgz',
+				'extract': True
+			})
+
+			# Set the artifact prefix for the browsertime results
+			job.setdefault('attributes', {})
+			job['attributes']['artifact_prefix'] = 'public/test_info'
+
+			# vismet runs on Linux but we want to have it displayed
+			# alongside the job it was triggered by to make it easier for
+			# people to find it back.
+			job['label'] = LABEL % {'platform': platform, 'label': dep_job.label}
```

Comment:
Nit: use `LABEL.format(platform=platform, label=dep_job.label)` and ` LABEL = "test-vismet-{platform}-{label}"`. Same thing for `SYMBOL`